### PR TITLE
add url support

### PR DESCRIPTION
**Pull request by hhsnopek**

### CODE REVIEW

**Comment by kylemac** [comment]
if we do this, wouldn't we want to remove the `name` opt all together, since it assumes `*.rooftoptopcms.io`?

**Comment by hhsnopek** [comment]
we could, tho that would be breaking change - I did this as a majority of users will probably end up sticking with `rooftopcms.io` as their host

**Comment by kylemac** [comment]
i don't mind a breaking change. i'd rather we just land on a good interface

**Comment by kylemac** [comment]
@jescalan and @hhsnopek discuss what would be the best interface, and come to a conclusion please

**Comment by jescalan** [comment]
Ah yeah I didn't consider the self-hosted option originally since I think we might be the first people to have actually set it up self-hosted haha. I think removing the name option is probably the way to go, as kyle suggested.

**Comment by hhsnopek** [comment]
I'm good with that 😄

**Comment by hhsnopek** [comment]
@jescalan updated to reflect the new interface - lmk what you think and if it's all good I'll rebase

File: test/index.js [hunk 1]
```diff
@@ -55,3 +55,21 @@ test('params passed to get() work correctly', (t) => {
     t.is(res[0].slug, 'hello-world')
   })
 })
+
+test('uses (http) url instead of name', (t) => {
```
Comment (jescalan) [inline test/index.js]: is this test redundant now?

Comment (hhsnopek) [inline test/index.js]: I think it could be redundant, but it also verifies that `http` is available if the protocol logic were to ever change

**Comment by hhsnopek** [comment]
@jescalan this is failing due to travis env variables but locally passes
local tests

File: README.md [hunk 2]
```diff
@@ -18,8 +18,8 @@ node api client for [Rooftop CMS](https://www.rooftopcms.com/)
 const Rooftop = require('rooftop-client')
 
 const api = Rooftop.new({
-  name: 'yoursubdomain',
-  apiToken: 'xxx'
+  url: 'https://yoursubdomain.rooftopcms.io'
+  apiToken: 'xxx',
```
Comment (jescalan) [inline README.md]: syntax error here

File: test/index.js (hunk 1 above)
Comment (jescalan) [inline test/index.js]: let's update the name to reflect this then

File: README.md (hunk 2 above)
Comment (hhsnopek) [inline README.md]: nice catch - fixed

**Comment by hhsnopek** [comment]
`url` was added to travis - I'll remove `name` once this PR is merged

File: test/index.js (hunk 1 above)
Comment (hhsnopek) [inline test/index.js]: 👍 updated

**Comment by edtjones** [comment]
Just to say you're all amazing @hhsnopek @kylemac @jescalan - thankyou for your efforts on this. Incidentally this is the interface we have on the ruby lib, so that's a nice byproduct of this PR. https://github.com/rooftopcms/rooftop-ruby#configuration

File: test/.env.sample [hunk 3]
```diff
@@ -1,2 +1,2 @@
-name=xxx
+url=htts://xxx
```
Comment (jescalan) [inline test/.env.sample]: this seems like it might be a typo?

Comment (hhsnopek) [inline test/.env.sample]: yup, fixed

**Comment by jescalan** [comment]
Thanks for sticking with it -- this looks great 👍

**Comment by hhsnopek** [comment]
I was totally gonna rebase this before you merged 😛

**Pull request merged**